storage/engine/mvcc.go

```diff
@@ -860,7 +860,14 @@ func MVCCConditionalPut(engine Engine, ms *MVCCStats, key proto.Key, timestamp p
 		return err
 	}
 
-	if expValue == nil && existVal != nil {
+	if _, _, err := MVCCGet(engine, key, timestamp, true /* consistent */, nil); txn != nil && existVal != nil && bytes.Equal(value.Bytes, existVal.Bytes) &&
```

Comment:
I hate to see this case added here. Could you at least put the comment outside the if block? And rearrange the logic more conventionally so it's not so fancily all being done in a single line?

Actually, better yet would be to simply ignore this case. Is it handled to make a unittest pass?